spark/src/main/java/com/netflix/iceberg/spark/source/Writer.java

```diff
@@ -309,10 +339,19 @@ private static String pathContext(Path dataPath) {
         String filename,
         FileFormat format,
         AppenderFactory<InternalRow> factory,
-        FileIO fileIo) {
+        FileIO fileIo,
+        EncryptionManager encryption,
+        boolean encryptionEnabled) {
       this.file = String.format("%s/%s", dataPath, filename);
       this.fileIo = fileIo;
-      this.appender = factory.newAppender(fileIo.newOutputFile(file), format);
+      OutputFile outputFile = fileIo.newOutputFile(file);
+      if (encryptionEnabled) {
+        maybeEncryptedOutput = encryption.encrypt(outputFile);
```

Comment:
I think I would prefer an invariant, where there is always an encryption manager and it may be a PlainTextEncryptionManager that returns a null key_metadata and the original stream.

Reply:
Actually, this isn't a big deal. An alternative is to use a null encryption manager instead of `encryptionEnabled`. If we have a separate boolean, then it should be something that can be overridden with write options, not just table options.